### ISSUE TITLE
Clean out broken links from the database?

### ISSUE THREAD

**Issue by petterreinholdtsen** [freetuxtv/freetuxtv#202]
Hi.  When I try to use freetuxtv, it is hard to find a channel/stream that is working.  Most are not working at all.  Is the database containing obsolete information?

I wrote a small script to test it, and a limited test show that 9 of 10 links were broken, while only 1 was working.  This is a similar script to test the entire database:

```
#!/bin/sh
# Count the number of non-working streams in the freetuxtv database

groupxml=/usr/share/freetuxtv/channels_groups.xml
ok=0
err=0
for dirurl in $(grep '<channels_group ' $groupxml | cut -d\" -f2); do
    for url in $(GET $dirurl |grep -v '#'); do
	mplayer -ss 0:00 -frames 1 -vo jpeg -nosound "$url"
	if [ ! -f 00000001.jpg ] ; then
	    echo "error $url"
	    err=$(($err + 1))
	else
	    rm 00000001.jpg
	    echo "ok $url"
	    ok=$(($ok + 1))
	fi
    done
done

echo "ok $ok failed $err"
```

**Comment by SaiyanRiku** [comment]
Thank for the script, i already thinking in making a program that test all link and report the status in the database directly. I wiil try to find time with your script to clean some link

**Comment by petterreinholdtsen** [comment]
Did you get the script I created working?  The database seem to still have lots of invalid entries, and I wonder what more can be done to help get rid of the broken data?

**Comment by SaiyanRiku** [comment]
Yes i tested it, but it's not working for some links. The script is blocking for some stream (like RTMP streams...). I quickly search for a timeout but i don't find the option in mplayer. I started to cleanup the database manually but it's very long.

**Comment by petterreinholdtsen** [comment]
What about using it for the links it can handle, to automatically clean out at least those broken URLs?

**Comment by petterreinholdtsen** [comment]
Hi.  Any news on this?  I still find urls in the database that do not work.

**Comment by SaiyanRiku** [comment]
I'm sorry i did not found time to work on it for now, due to professional projects. I need to update the manager to check every link posted because there is a lot of spam with bad url.

**Comment by RealVegOs** [comment]
What about a method to add own URLs? Needs only a blank entry with properties dialogue to be cloned. How to submit URLs (sorry I did not check all issues)? Alas due to skills, I could not help with programming. Some URLs suck, like Juvelo TV and Russia Today, but ... It's about tech here.

**Comment by SaiyanRiku** [comment]
URL can be submitted in database.freetuxtv.net, if someone have time and want to make moderation on URL to sort them, i can create an account in the manager to get privilege to change the stream status.

**Comment by Shridha12** [comment]
Will i ask how to run this script?

**Comment by Padre2** [comment]
@Jet1oeil   Of the 12 channels in Web TV (English), there only appears to be 1 channel, France 24, that is working.  similar problem with other Web TV channels.  Most do not even give an error message. Is it not possible to remove all the dead channels, or should we assume that this Freetux tv is no longer maintained.

**Comment by freddii** [comment]
> URL can be submitted in database.freetuxtv.net, if someone have time and want to make moderation on URL to sort them, i can create an account in the manager to get privilege to change the stream status.

I like to add streams and change stream status.

**Comment by SaiyanRiku** [comment]
> > URL can be submitted in database.freetuxtv.net, if someone have time and want to make moderation on URL to sort them, i can create an account in the manager to get privilege to change the stream status.
> 
> I like to add streams and change stream status.

Sorry for the response delay. If you want an account, please send me a mail to eric.beuque [at] gmail.com, with your required username and password.
